File: 1.1/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceMultisamplePropertiesEXT.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceMultisamplePropertiesEXT extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceMultisamplePropertiesEXT() {
    }

    protected PFNvkGetPhysicalDeviceMultisamplePropertiesEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceMultisamplePropertiesEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceMultisamplePropertiesEXT(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceMultisamplePropertiesEXT(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceMultisamplePropertiesEXT"));
    }

    public void call(VkPhysicalDevice physicalDevice, int samples, VkMultisamplePropertiesEXT pMultisampleProperties){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), samples, pMultisampleProperties != null ? pMultisampleProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long physicalDevice, int samples, long pMultisampleProperties);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceMultisamplePropertiesEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint samples, jlong pMultisampleProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT f = (PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT)jniLongToFunctionPointer(address);
    f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (VkSampleCountFlagBits)samples,
        ((VkMultisamplePropertiesEXT*)jniLongToPointer(pMultisampleProperties))
    );
}

